pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/forward/FixedBytePower2ChunkSVForwardIndexReader.java

```diff
@@ -111,4 +115,43 @@ protected ByteBuffer getChunkBuffer(int docId, ChunkReaderContext context) {
     }
     return decompressChunk(chunkId, context);
   }
+
+  protected void recordDocIdRanges(int docId, ChunkReaderContext context, List<ValueRange> ranges) {
+    int chunkId = docId >>> _shift;
+    if (context.getChunkId() == chunkId) {
+      ranges.addAll(context.getRanges());
+      return;
+    }
+    recordChunkRanges(chunkId, context, ranges);
+  }
+
+  @Override
+  public List<ValueRange> getDocIdRange(int docId, ChunkReaderContext context, @Nullable List<ValueRange> ranges) {
+    if (ranges == null) {
+      ranges = new ArrayList<>();
+    }
+    if (_isCompressed) {
+      recordDocIdRanges(docId, context, ranges);
+    } else {
+      // If uncompressed, should use fixed offset
+      throw new IllegalStateException("Operation not supported since the forward index is of fixed length type");
+    }
+
+    return ranges;
+  }
+
+  @Override
+  public boolean isFixedLengthType() {
+    return !_isCompressed;
+  }
+
+  @Override
+  public long getBaseOffset() {
+    return _rawDataStart;
+  }
+
+  @Override
+  public int getDocLength() {
+    return _storedType.size();
+  }
```

Comment:
same here: should we throw exception when `isFixedLengthType()` returns false?

Reply:
Ack